File: Excel/400-499/473/473 Challenge.R
print_banner_matrix <- function(word) {
  n <- nchar(word)
  banner_matrix <- matrix("", nrow = 4, ncol = n * 3)
  for (i in seq_len(n)) {
    char <- substr(word, i, i)
    if (char != " ") {
      banner_matrix[1, (i - 1) * 3 + 1:3] <- c(" ", "_", " ")
      banner_matrix[2, (i - 1) * 3 + 1:3] <- c("/", " ", "\\")
      banner_matrix[3, (i - 1) * 3 + 1:3] <- c("|", char, "|")
      banner_matrix[4, (i - 1) * 3 + 1:3] <- c("\\", "_", "/")
    } else {
      banner_matrix[1, (i - 1) * 3 + 1:3] <- rep(" ", 3)
      banner_matrix[2, (i - 1) * 3 + 1:3] <- rep(" ", 3)
      banner_matrix[3, (i - 1) * 3 + 1:3] <- rep(" ", 3)
      banner_matrix[4, (i - 1) * 3 + 1:3] <- rep(" ", 3)
    }
  }
  apply(banner_matrix, 1, function(row) cat(paste(row, collapse = ""), "\n"))
}

print_banner_matrix("EXCEL")
print_banner_matrix("MICROSOFT")
print_banner_matrix("POWER QUERY")

File: Excel/400-499/473/473 Challenge.py
def print_banner_matrix(word):
    n = len(word)
    banner_matrix = [["" for _ in range(n * 3)] for _ in range(4)]
    for i in range(n):
        char = word[i]
        if char != " ":
            banner_matrix[0][(i * 3):(i * 3) + 3] = [" ", "_", " "]
            banner_matrix[1][(i * 3):(i * 3) + 3] = ["/", " ", "\\"]
            banner_matrix[2][(i * 3):(i * 3) + 3] = ["|", char, "|"]
            banner_matrix[3][(i * 3):(i * 3) + 3] = ["\\", "_", "/"]
        else:
            banner_matrix[0][(i * 3):(i * 3) + 3] = [" "] * 3
            banner_matrix[1][(i * 3):(i * 3) + 3] = [" "] * 3
            banner_matrix[2][(i * 3):(i * 3) + 3] = [" "] * 3
            banner_matrix[3][(i * 3):(i * 3) + 3] = [" "] * 3
    for row in banner_matrix:
        print("".join(row))

print_banner_matrix("EXCEL")
print_banner_matrix("MICROSOFT")
print_banner_matrix("POWER QUERY")